File: cases/taint/c/element-object-negative/element_object_negative.c
struct Item {
    int value;
};

int dfb_source(void) { // DFB-SOURCE: element-object-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: element-object-sink

void run(void) {
    struct Item items[2];
    items[0].value = dfb_source(); // DFB-WITNESS: element-object-store
    items[1].value = 0;
    dfb_sink(items[1].value);
}

File: cases/taint/scala/element-object-negative/ElementObjectNegative.scala
package dataflowbench

object ElementObjectNegative {
  class Item {
    var value: String = "clean"
  }

  def dfb_source(): String = { // DFB-SOURCE: element-object-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: element-object-sink

  def run(): Unit = {
    val items = Array(new Item(), new Item())
    items(0).value = dfb_source() // DFB-WITNESS: element-object-store
    items(1).value = "clean"
    dfb_sink(items(1).value)
  }
}
